app/scenes/channel/channel_post_list/channel_post_list.js

```diff
@@ -116,15 +118,25 @@ export default class ChannelPostList extends PureComponent {
 
     render() {
         const {
+            actions,
             applicationInitializing,
             channel,
             channelIsLoading,
             posts,
             postsRequests,
-            theme
+            theme,
+            networkOnline
         } = this.props;
+
         let component;
-        if (!applicationInitializing && !channelIsLoading && posts && (postsRequests.getPosts.status !== RequestStatus.STARTED || !this.state.didInitialPostsLoad)) {
+        if (!this.state.didInitialPostsLoad && !posts.length && channel.total_msg_count > 0 && (postsRequests.getPosts.status !== RequestStatus.STARTED || !networkOnline)) {
```

Comment:
No, it should be `||` since we want to show the component if we know that the channel has posts but none we're received because the the request failed and we're still connected.